src/buildstream/_frontend/widget.py

```diff
@@ -437,6 +437,14 @@ def show_pipeline(self, dependencies, format_):
                 runtime_deps = [e._get_full_name() for e in element._dependencies(_Scope.RUN, recurse=False)]
                 line = p.fmt_subst(line, "runtime-deps", _yaml.roundtrip_dump_string(runtime_deps).rstrip("\n"))
 
+            # CAS Digest
+            if "%{cas-digest" in format_:
+                cas_digest = element._get_cas_digest()
```

Comment:
I kinda think we should implement this here for the moment. Call `element._get_artifact()` to get the artifact, and try to format the output here.

Later, we could have a better API to expose to the plugins, and we can use it here too.